docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -837,15 +831,458 @@ spec:
+
+[id="{p}-logstash-inputs-data-pushed"]
+==== Input plugins: events pushed to {ls}
+
+{ls} installations with inputs that enable {ls} to receive data should be able to scale freely and have load spread across them horizontally.
+These plugins include {logstash-ref}/plugins-inputs-beats.html[`logstash-input-beats`], {logstash-ref}/plugins-inputs-elastic_agent.html[`logstash-input-elastic_agent`],  {logstash-ref}/plugins-inputs-tcp.html[`logstash-input-tcp`], and {logstash-ref}/plugins-inputs-http.html[`logstash-input-http`].
+
+[id="{p}-logstash-inputs-local-checkpoints"]
+==== Input plugins: {ls} maintains state 
+
+{ls} installations that use input plugins that retrieve data from an external source, and **maintain local checkpoint state**, or would require some level of co-ordination between nodes to split up work can specify `pipeline.workers` freely, but should keep the pod count at 1 for each {ls} installation.
+
+Note that plugins that retrieve data from external sources, and require some level of coordination between nodes to split up work, are not good candidates for scaling horizontally, and would likely produce some data duplication. 
+
+Input plugins that include configuration settings such as  `sincedb`, `checkpoint` or `sql_last_run_metadata` may fall into this category.
+
+Examples of these plugins include {logstash-ref}/plugins-inputs-jdbc.html[`logstash-input-jdbc`] (which has no automatic way to split queries across {ls} instances), {logstash-ref}/plugins-inputs-s3.html[`logstash-input-s3`] (which has no way to split which buckets to read across {ls} instances), or {logstash-ref}/plugins-inputs-file.html[`logstash-input-file`].
+
+[id="{p}-logstash-inputs-external-state"]
+==== Input plugins: external source stores state
+
+{ls} installations that use input plugins that retrieve data from an external source, and **rely on the external source to store state** can scale based on the parameters of the external source.
+
+For example, a {ls} installation that uses a {logstash-ref}/plugins-inputs-kafka.html[`logstash-input-kafka`] plugin to retrieve data can scale the number of pods up to the number of partitions used, as a partition can have at most one consumer belonging to the same consumer group. 
+Any pods created beyond that threshold cannot be scheduled to receive data.
+
+Examples of these plugins include {logstash-ref}/plugins-inputs-kafka.html[`logstash-input-kafka`], {logstash-ref}/plugins-inputs-azure_event_hubs.html[`logstash-input-azure_event_hubs`], and {logstash-ref}/plugins-inputs-kinesis.html[`logstash-input-kinesis`].
+
+[id="{p}-logstash-working-with-plugin-considerations"]
+=== Plugin-specific considerations
```

Comment:
I added it.  Note that I also reformatted the section to use headings instead of the definition list format. Please check to see if you're OK with the description I used before your example, or if you would like for me to tweak it.

Reply:
Please see if I implemented this as you pictured it.

Reply:
btw, the source code in the [original note](https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-logstash-configuration.html#k8s-logstash-pipelines-es) is missing `]` at the end and having an extra `,` after `"cluster:admin/ingest/pipeline/get"` which will cause syntax error :)

This is the correct yaml
```yaml
"cluster": ["monitor", "manage_ilm", "read_ilm", "manage_logstash_pipelines", "manage_index_templates", "cluster:admin/ingest/pipeline/get"],
"indices": [
  {
    "names": [ "logstash", "logstash-*", "ecs-logstash", "ecs-logstash-*", "logs-*", "metrics-*", "synthetics-*", "traces-*" ],
    "privileges": ["manage", "write", "create_index", "read", "view_index_metadata"]
  }
]
```